### DISCUSSION TITLE
QR code

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco.UI#650] [Ideas]
Maybe it would be great to have a QR code component in the UI library?
Something similar to this: https://shoelace.style/components/qr-code

I don't recall in the current backoffice shows a QR code in the 2FA view, but I recall we had a backoffice training many years ago when Umbraco 7 was 🔥 .. adding a custom dashboard showing a QR code.
<img src="https://github.com/umbraco/Umbraco.UI/assets/2919859/c0bcb49e-15ae-4824-9b0e-d9baabfe840b" width="128" height="128">

**Comment by bjarnef** [comment]
@iOvergaard @nielslyngsoe I think this would make sense to include in UI library as it is used in Two-factor authentication:
https://docs.umbraco.com/umbraco-cms/reference/security/two-factor-authentication

I also noticed is is used with Umbraco Id on Umbraco Cloud.

**Reply by iOvergaard** [reply]
The QR-code used in two-factor authentication in Umbraco is generated by whatever authentication library, you are using. Google, for instance, will send an image back to you containing a QR code, so there is no need to generate one manually. I am not even sure what such a QR code contains.

It's not a bad idea to have a component like this (which seems to primarily generate links), but it's not something we can use for 2FA.

**Comment by bjarnef** [comment]
Another example: https://hackernoon.com/crafting-interactive-qr-code-web-components-integration-and-customization

which use [QR Code JS](https://github.com/danielgjackson/qrcodejs?ref=hackernoon.com)
